example-cropping/README.md

```diff
@@ -0,0 +1,779 @@
+  return appwrite
+    .setEndpoint(ep)
+    .setProject(pi)
+
+}
+export { appWrite };
+
+```
+
+
+## How To Use Cropping Using Appwrite SDK
+
+To crop an image using appwrite SDK is actually pretty simple you just need to upload your file into appwrite storage then preview the image and pass the optional parameters, if you take a look at the js docs in https://appwrite.io/docs/client/storage#getFilePreview you will see all of the optional parameters and how to use it. Below is the example of how to use it using js SDK
+
+```js
+let sdk = new Appwrite();
+
+sdk
+    .setEndpoint('https://[HOSTNAME_OR_IP]/v1') // Your API Endpoint
+    .setProject('5df5acd0d48c2') // Your project ID
+;
+
+let result = sdk.storage.getFilePreview('[FILE_ID]',300,null,100,null,"webp");
+
+console.log(result); // Resource URL for example http://localhost/v1/storage/files/5f70694f8637d/preview?width=300&quality=100&output=webp&project=5f6f49028c5b9
+
+```
+
+using the above script we will get our image with the output that we like so for example we pass 300 as width,100 as quality and output webp we will get exactly the output like that without the need of cdn to change the width, height, quality ,background, and output the appwrite will do it just like we want. Now we will create a component that can implement this functionality using react, below is step by step how to create simple app that can crop our image from appwrite storage starting from signup, login, upload, list, then preview and crop.
+
+## Step 1 - Create PreviewImage Component
+
+So this is actually the main ingredient in our tutorial what this does is to show the preview of the image in our server and return in a spesific parameter that we want so in order to use this component we can pass the parameter exactly the same as in the appwrite SDK documentation which is id,width,height,quality,background,and output and it will output the image with that spesific condition which mean this component is used for cropping an image that we get from our storage.
+
+```js
+import React, { useState } from 'react';
+
+function PreviewImage(props) {
+
+  function getImage(id){
+
+    let mWidth=props.width || null;
+    let mHeight=props.height || null;
+    let mQuality=props.quality || 100;
+    let mBackground=props.background || null;
+    let mOutput=props.output || null;
+    let image=props.appwrite.storage.getFilePreview(id,mWidth,mHeight,mQuality,mBackground,mOutput);
+
+
+    return image
+  }
+
+    return (<img src={props.id?getImage(props.id):""} />)
+  
+};
+
+export { PreviewImage };
+```
+
+example usage
+
+```js
+<PreviewImage appwrite={props.appwrite} width={width} height={height} quality={quality} background={background} output={output} id={props.imageId} />
+```
+
+as you can see there is so many props there to use it actually the same as in the SDK but instead we implement it in react the different maybe there is ```appwrite``` props which is used for interacting with appwrite SDK
```

Comment:
Hmm. I'm not sure I understood the meaning of this line

Reply:
actually the props there is the same as parameter in appwrite SDK 
![image](https://user-images.githubusercontent.com/62529025/95676992-b01fe600-0bec-11eb-9700-124cade66a44.png)
maybe the word should be rewrited let me rewrite it

Reply:
```suggestion
the props that you can use in this component is actually the same as in the appwrite SDK which is id, width, height, quality, background, and output you can check the function in the docs [here](https://appwrite.io/docs/client/storage#getFilePreview)
```

how about this one?

Reply:
```suggestion
The props that you can use in this component are actually the same as in the Appwrite SDK which is id, width, height, quality, background, and output. You can check the function in the docs [here](https://appwrite.io/docs/client/storage#getFilePreview)
```

This is better